README.md

```diff
@@ -6,82 +6,94 @@
-* __StandardizeReads__: Standardize quality and format for all kind of sources for ReadTools
-* __ReadsToFastq__: Converts any kind of ReadTools source to FASTQ format
-* __AssignReadGroupByBarcode__: Assigns read groups based on barcode tag(s) for all kind of sources for ReadTools
-* __TrimReads__: Applies a trimming pipeline to any kind of sources for ReadTools
 
-Legacy tools that will disapear in following releases:
-* __QualityChecker__: Get the quality encoding for a BAM/FASTQ file
-* __StandardizeQuality__: Convert an Illumina BAM/FASTQ file into a Sanger
-* __TaggedBamToFastq__: Convert an BAM file with BC tags into a FASTQ file
-* __FastqBarcodeDetector__: Identify barcodes in the read name for a FASTQ file and assign to the ones used on the library
-* __BamBarcodeDetector__:	Identify barcodes in the read name for a BAM file and assign to the ones used on the library
-* __TrimFastq__: Implementation of the trimming algorithm from [Kofler _et al._ (2011)](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0015925)
+## Testing
 
----
+Code coverage reports for the project can be found in the [Codecov project page](https://codecov.io/gh/magicDGS/ReadTools).
+If you want to evaluate _ReadTools_ locally:
+* To run all tests, run `./gradlew test`. Test reports will be in __build/reports/tests/index.html__
+* To compute a coverage report, run `./gradlew jacocoTestReport`. The report will be in __build/reports/jacoco/test/html/index.html__
 
-### Installation from source
+## Guidelines
 
-Master branch is guarantee to compile successfully and contain the latest release changes. For installation, run:
+* __Pushing directly to master branch is not allowed.__
+* It is recommended to name branches with a short form of your name and a explanatory name. Example: dgs_fix_issue30.
+* Pull Requests should be reviewed before merging by other developer.
+* Any new code will require unit/integration tests.
+* Use [org.apache.logging.log4j.Logger](https://logging.apache.org/log4j/2.0/log4j-api/apidocs/org/apache/logging/log4j/Logger.html) for logging.
+* Use [TestNG](http://testng.org/doc/index.html) for testing.
+* Use [magicDGS Style Guide](https://github.com/magicDGS/styleguide) for code formatting.
 
-```
+## Versioning and changelog
 
-git clone https://github.com/magicDGS/ReadTools.git
-cd ReadTools
-./gradlew shadowJar
+We use [semantic versioning](http://semver.org/) (MAJOR.MINOR.PATCH).
+[build.gradle] and Unreleased tag (within [CHANGELOG]) should be updated accordingly in Pull Request to follow the convention:
 
-```
+* Backwards-compatible bug fixes: increase PATCH number. Add a __Fixed__ entry to [CHANGELOG].
+* Backwards-compatible changes: increase MINOR number and reset PATCH to 0. Add a __Added__ entry to [CHANGELOG].
+* Backwards-incompatible changes: increase MAJOR number and reset MINOR and PATCH to 0. Add a __Changed__ entry to [CHANGELOG].
```

Comment:
Only increase the MAJOR number if it is the same as the previously released version. If it is already changed, apply either the first or the second depending on the change.

This will ensure that the major version is not growing crazily.